pkg/apis/core/validation/validation.go

```diff
@@ -2228,6 +2237,19 @@ func ValidatePersistentVolumeClaimSpec(spec *core.PersistentVolumeClaimSpec, fld
 				"must match dataSourceRef"))
 		}
 	}
+	if spec.VolumeAttributesClassName != nil {
+		if opts.EnableVolumeAttributesClass {
+			if len(*spec.VolumeAttributesClassName) == 0 {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("volumeAttributesClassName"), *spec.VolumeAttributesClassName, "an empty string is disallowed"))
+			} else {
+				for _, msg := range ValidateClassName(*spec.VolumeAttributesClassName, false) {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("volumeAttributesClassName"), *spec.VolumeAttributesClassName, msg))
+				}
+			}
+		} else {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("volumeAttributesClassName"), "cannot be specified unless the VolumeAttributesClass feature gate is enabled"))
```

Comment:
Would we ever reach here considering that the field gets dropped in the strategy?